src/site/content/en/blog/gamepad/index.md

```diff
@@ -0,0 +1,348 @@
+  // the game loop, not outside.
+  const gamepads = navigator.getGamepads();
+  for (const gamepad of gamepads) {
+    // Disregard empty slots.
+    if (!gamepad) {
+      continue;
+    }
+    // Process the gamepad state.
+    console.log(gamepad);
+  }
+  // Call yourself upon the next animation frame.
+  // (Typically this happens every 60 times per second.)
+  window.requestAnimationFrame(pollGamepad)
+};
+// Kick off the initial game loop iteration.
+pollGamepad();
+```
+
+{% Aside 'gotchas' %}
+  Do not store a lasting reference to the `GamepadList` result *outside* of the game loop,
+  since the method returns a static snapshot, not a live object.
+  Call `navigator.getGamepads()` each time anew *in your game loop*.
+{% endAside %}
+
+### Making use of the vibration actuator
+
+The `vibrationActuator` property returns a `GamepadHapticActuator`
+object, which corresponds to a configuration of motors or other actuators
+that can apply a force for the purposes of haptic feedback.
+Haptic effects can be played by calling `Gamepad.vibrationActuator.playEffect()`.
+The only currently valid effect type is `'dual-rumble'`.
+Dual-rumble describes a haptic configuration with an eccentric rotating mass vibration motor
+in each handle of a standard gamepad.
+In this configuration, either motor is capable of vibrating the whole gamepad.
+The two masses are unequal so that the effects of each can be combined
+to create more complex haptic effects.
+Dual-rumble effects are defined by four parameters:
+
+- `duration`:
+   Sets the duration of the vibration effect in milliseconds.
+- `startDelay`:
+  Sets the duration of the delay until the vibration is started.
+- `strongMagnitude` and `weakMagnitude`:
+  Set the vibration intensity levels for the heavier and lighter eccentric rotating mass
+  motors, normalized to the range `0.0`–`1.0`.
+
+```js
+// This assumes a `Gamepad` as the value of the `gamepad` variable.
+const vibrate = (gamepad, delay = 0, duration = 100, weak = 1.0, strong = 1.0) {
+  if (!('vibrationActuator' in gamepad)) {
+    return;
+  }
+  gamepad.vibrationActuator.playEffect('dual-rumble', {
+    // Start delay in ms.
+    startDelay: delay,
+    // Duration is ms.
+    duration: duration,
+    // The magnitude of the weak actuator (between 0 and 1).
+    weakMagnitude: weak,
+    // The magnitude of the strong actuator (between 0 and 1).
+    strongMagnitude: strong,
+  });
+};
+```
+
+### Integration with Permissions Policy
+
+The Gamepad API spec defines a
+[policy-controlled feature](https://w3c.github.io/webappsec-permissions-policy/)
+identified by the string `"gamepad"`. Its default `allowlist` is `"self"`.
+A document's permissions policy determines whether any content in that document is allowed
+to access `navigator.getGamepads()`.
+If disabled in any document, no content in the document will be allowed to use
+`navigator.getGamepads()`, nor will the `gamepadconnected` and
+`gamepaddisconnected` events fire.
+
+```html/2
```

Comment:
I'm not sure having lines highlighted helps. One line would be enough for me.

Reply:
Indeed, it's really short enough.